code/obj/sawflymisc.dm

```diff
@@ -0,0 +1,630 @@
+
+		if (alreadydead) return // prevents any death behavior from ever happening more than once
+
+		alive = FALSE
+		src.set_density(0)
+		src.force_laydown_standup()
+		src.tokenized_message(death_text)
+		src.is_npc = FALSE // stop any and all possible non-critter AI thought
+		src.throws_can_hit_me = FALSE  //prevent getting hit by thrown stuff- super important in avoiding jank
+
+
+		animate(src) //stop no more float animation
+		icon_state = "sawflydead[pick("1", "2", "3", "4", "5", "6", "7", "8")]" //randomly selects death icon and displaces them
+		src.pixel_x += rand(-5, 5)
+		src.pixel_y += rand(-1, 5)
+		src.anchored = 0
+		walk_to(src,0) //halt walking
+
+		// special checks that determine how much damage they do after death
+		if (prob(20))
+			new /obj/item/device/prox_sensor(src.loc)
+			return
+
+		if(prob(60))
+			elecflash(src, 1, 3)
+
+		if(prob(20)) // congrats, little guy! You're special! You're going to blow up!
+			if(prob(70)) //decide whether or not people get a warning
+				src.visible_message("<span class='combat'>[src] makes a [pick("gentle", "odd", "slight", "weird", "barely audible", "concerning", "quiet")] [pick("hiss", "drone", "whir", "thump", "grinding sound", "creak", "buzz", "khunk")].......")
+			SPAWN(deathtimer SECONDS)
+				src.blowup()
+		alreadydead = TRUE
+
+
+
+	proc/blowup() // used in emagged controllers and has a chance to activate when they die
+
+		if(prob(66))
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "fuel tank", "battery", "thruster")] [pick("combusts", "catches on fire", "ignites", "lights up", "bursts into flames")]!")
+			fireflash(src,1,TRUE)
+		else
+			src.visible_message("<span class='combat'>[src]'s [pick("motor", "core", "head", "engine", "thruster")] [pick("overloads", "blows up", "catastrophically fails", "explodes")]!")
+			fireflash(src,0,TRUE)
+			explosion(src, get_turf(src), 0, 1, 1.5, 3)
+			qdel(src)
+
+		if(alive) // prevents weirdness from emagged controllers causing frankenstein sawflies
+			qdel(src)
+
+// FROM HERE ON OUT IS BEHAVIOR SHIT, MOST OF WHICH IS PORTED FROM CRITTERS
+
+	proc/ChaseAttack(atom/M)
+		if (istraitor(M) || isnukeop(M) || isspythief(M) ||(M in src.friends))
+			return
+		if(target && !attacking)
+			attacking = TRUE
+			src.visible_message("<span class='alert'><b>[src]</b> flies at [M]!</span>")
+			if(prob(25)) communalbeep()
+			if (istraitor(M) || isnukeop(M) || isspythief(M) || (M in src.friends))
+				return
+			task = "chasing"
+			walk_to(src, src.target,1,4)
+			var/tturf = get_turf(M)
+			Shoot(tturf, src.loc, src)
+			SPAWN(attack_cooldown)
+				attacking = FALSE
+			return
+
+
+	proc/seek_target()
+		src.anchored = FALSE
+
+		for (var/mob/living/C in view(src.seekrange,src)) //search for secoffs before targeting any crew
+			if(C == oldtarget_name) continue
+			if (C.health < -50) continue
+			if (C.job == "Security Officer" || C.job == "Head of Security")
```

Comment:
Them magically knowing this doesn't make much sense, should go via ID if anything.